include/structures/PriorityQueue.hpp

```diff
@@ -0,0 +1,53 @@
+/*
+ * PriorityQueue.hpp
+ *
+ * Created on: 29.03.2025
+ * Author: Jan Kukowski
+ */
+
+#pragma once
+
+#include <functional>
+
+namespace Koala {
+
+/**
+ * @ingroup priority-queue
+ * The base class for priority queues.
+ */
+template <class Key, class Compare = std::less<Key>>
+class PriorityQueue {
+public:
+    /**
+     * Removes and returns the top element of the priority queue.
+     * 
+     * @return Key The element with the highest priority.
+     * @throws std::runtime_error If the priority queue is empty.
+     */
+    virtual Key pop() = 0;
+
+    /**
+     * Returns the top element of the priority queue without removing it.
+     * 
+     * @return Key The element with the highest priority.
+     * @throws std::runtime_error If the priority queue is empty.
+     */
+    virtual Key peek() const = 0;
```

Comment:
Just for consistency with std: please make this `top` not `peek`, and let the value returned be `const Key&` (why virtual?)

Reply:
fixed, it is only pure virtual to force implementation in derived classes